.github/workflows/check_ci.yaml

```diff
@@ -80,6 +87,14 @@ jobs:
      - name: Setup Python & Poetry Environment
        uses: ./.github/actions/prepare_poetry_env
 
+     - name: Run Unit Tests
```

Comment:
create new job and make the integration and notebook tests depend on it with needs_to

Reply:
See next push